llvm/lib/Analysis/ConstantFolding.cpp

```diff
@@ -2422,6 +2423,21 @@ static Constant *ConstantFoldScalarCall1(StringRef Name,
 
       return ConstantFP::get(Ty->getContext(), Val);
     }
+
+    case Intrinsic::amdgcn_s_wqm: {
+      uint64_t Val = Op->getZExtValue();
+      uint64_t WQM = 0;
+      uint64_t Quad = 0xF;
+      for (unsigned i = 0; i < Op->getBitWidth() / 4;
```

Comment:
Upper case `I`.

As an alternative, you could implement this one with just two rounds of bit twiddling:
```
Val |= (Val & 0x5555555555555555) << 1| ((Val >> 1) & 0x5555555555555555);
Val |= (Val & 0x3333333333333333) << 2| ((Val >> 2) & 0x3333333333333333);
```